shared/AppInsightsCore/src/Config/DynamicConfig.ts

```diff
@@ -62,7 +62,7 @@ function _createDynamicHandler<T extends IConfiguration>(logger: IDiagnosticLogg
     }
 
     function _setValue<C, V>(target: C, name: string, value: V) {
-        return _setDynamicProperty(theState, target, name, value);
+        return _setDynamicProperty(theState, target, name, value)[name];
```

Comment:
I changed the _setDynamicProperty() to return then `target` rather than the `target[name]` as that was indirectly causing the objects to be referenced and converted to be dynamic (if they where not already).

This caused an issue when "setting" the default values (which didn't use the returned value) and also indirectly caused everything to become dynamic during initialization whether referenced or not.